File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = [[0, i + 1] for i in range(N)]
    
    for i in range(1, N + 1):
        clear_person = len([item for item in stages if item >= i])
        not_clear_person = len([item for item in stages if item == i])
        if (not_clear_person == 0):
            answer[i - 1][0] = 0
        else:
            answer[i - 1][0] = not_clear_person / clear_person
    
    answer = sorted(answer, key = lambda x : x[0], reverse = True)
        
    return [num[1] for num in answer]

File: 프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    let result = Array(N).fill().map((item, index) => {
        const reachedUsers = stages.filter((stage) => stage >= index + 1);
        const notClearUsers = stages.filter((stage) => stage === index + 1);
        return [index + 1, notClearUsers.length / reachedUsers.length]
    })
    result = result.sort((a, b) => {
        if (a[1] === b[1]) {
            return a[0] - b[0];
        }
        return b[1] - a[1];
    }).map((item) => item[0])

    return result;
}